apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,120 @@ package managedidentity
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
 	"testing"
+	"time"
+
+	internalTime "github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/json/types/time"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/oauth/ops/accesstokens"
 )
 
-func fakeClient(mangedIdentityId ID, options ...Option) (Client, error) {
-	client, err := New(mangedIdentityId, options...)
+func fakeMIClient(mangedIdentityId ID, options ...ClientOption) (Client, error) {
+	fakeClient, err := New(mangedIdentityId, options...)
 
 	if err != nil {
 		return Client{}, err
 	}
 
-	return client, nil
+	return fakeClient, nil
+}
+
+type errorClient struct{}
+
+func (*errorClient) CloseIdleConnections() {}
+func (*errorClient) Do(req *http.Request) (*http.Response, error) {
+	return nil, fmt.Errorf("expected no requests but received one for %s", req.URL.String())
+}
+
+type FakeClient struct {
+	responseType int
+}
+
+func (c *FakeClient) CloseIdleConnections() {}
+func (c *FakeClient) Do(req *http.Request) (*http.Response, error) {
+	println(c.responseType)
+	w := makeResponse(c.responseType)
+	return &w, nil
+}
+
+func makeResponse(responseType int) http.Response {
```

Comment:
Consider creating separate methods for each response type. This param makes the code hard to read.